liblocation/src/main/java/com/mapbox/android/core/location/LocationPendingIntentProvider.java

```diff
@@ -0,0 +1,32 @@
+package com.mapbox.android.core.location;
+
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+
+import java.util.Random;
+
+class LocationPendingIntentProvider {
+
+  private final LocationPendingIntent locationPendingIntent;
+
+  LocationPendingIntentProvider(Context context, GoogleLocationEngine googleLocationEngine) {
+    locationPendingIntent = buildIntent(context, googleLocationEngine);
+  }
+
+  LocationPendingIntent intent() {
+    return locationPendingIntent;
+  }
+
+  static LocationPendingIntent buildIntent(Context context, GoogleLocationEngine googleLocationEngine) {
+    Intent intent = new Intent("GoogleLocationEngineBroadcast-" + googleLocationEngine.toString());
```

Comment:
We can pass the `action` string instead of building it from scratch with the engine's instance.